library/src/main/java/de/rheinfabrik/heimdall/OAuth2AccessTokenManager.java

```diff
@@ -71,17 +72,25 @@ public OAuth2AccessTokenManager(OAuth2AccessTokenStorage<TAccessToken> storage)
             throw new IllegalArgumentException("Grant MUST NOT be null.");
         }
 
-        return grant
-                .grantNewAccessToken()
-                .doOnSuccess(accessToken -> {
-                    if (accessToken.expiresIn != null) {
-                        Calendar expirationDate = (Calendar) calendar.clone();
-                        expirationDate.add(Calendar.SECOND, accessToken.expiresIn);
-                        accessToken.expirationDate = expirationDate;
-                    }
+        if (mTokenObservable == null) {
+            mTokenObservable = grant
+                    .grantNewAccessToken()
+                    .doOnSuccess(accessToken -> {
+                        if (accessToken.expiresIn != null) {
+                            Calendar expirationDate = (Calendar) calendar.clone();
+                            expirationDate.add(Calendar.SECOND, accessToken.expiresIn);
+                            accessToken.expirationDate = expirationDate;
+                        }
+
+                        mStorage.storeAccessToken(accessToken);
+
+                        mTokenObservable = null;
+                    })
+                    .toObservable()
+                    .cache();
```

Comment:
`cache` here would mean that every time we subscribe every element will be replayed right?